types/react/next.d.ts

```diff
@@ -27,4 +27,53 @@ import React = require('.');
 
 export {};
 
-declare module '.' {}
+declare module '.' {
+    interface ThenableImpl<T> {
+        then(onFulfill: (value: T) => unknown, onReject: (error: unknown) => unknown): void | PromiseLike<unknown>;
+    }
+    interface UntrackedThenable<T> extends ThenableImpl<T> {
+        status?: void;
+    }
+
+    export interface PendingThenable<T> extends ThenableImpl<T> {
+        status: 'pending';
+    }
+
+    export interface FulfilledThenable<T> extends ThenableImpl<T> {
+        status: 'fulfilled';
+        value: T;
+    }
+
+    export interface RejectedThenable<T> extends ThenableImpl<T> {
+        status: 'rejected';
+        reason: unknown;
+    }
+
+    export type Thenable<T> = UntrackedThenable<T> | PendingThenable<T> | FulfilledThenable<T> | RejectedThenable<T>;
+
+    export type Usable<T> = Thenable<T> | Context<T>;
+
+    export function use<T>(usable: Usable<T>): T;
+
+    interface ServerContextJSONArray extends ReadonlyArray<ServerContextJSONArray> {}
+    export type ServerContextJSONValue =
+        | string
+        | boolean
+        | number
+        | null
+        | ServerContextJSONArray
+        | { [key: string]: ServerContextJSONValue };
+    export interface ServerContext<T extends ServerContextJSONValue> {
+        Provider: Provider<T>;
+    }
+    function useContext<T extends ServerContextJSONValue>(context: ServerContext<T>): T;
```

Comment:
Does this add another overload or override? Might be worth a test that useContext still works with a plain Context too.

Reply:
I hope that it does. Adding a test to confirm (since we might need it again for adding more Usable types).

Reply:
Added a test and it does work as intended. VSCode also says that this works as an overload (`useContext` from stable release has no overloads):
![Screenshot 2022-10-24 at 15 04 49](https://user-images.githubusercontent.com/12292047/197532429-31198e1d-33b3-4023-b2cb-2c1961516c46.png)